semgrep-core/src/engine/Match_rules.ml

```diff
@@ -986,33 +994,23 @@ let check hook default_config rules equivs file_and_more =
                logger#info "skipping rule %s for %s" r.R.id file;
                RP.empty_semgrep_result )
              else
-               let xpatterns = xpatterns_in_formula formula in
+               let formula = Rule.formula_of_rule r in
                let config = r.settings ||| default_config in
-               let res =
-                 matches_of_xpatterns config equivs
-                   (file, xlang, lazy_ast_and_errors, lazy_content)
-                   xpatterns
-               in
-               logger#info "found %d matches" (List.length res.matches);
-               (* match results per minirule id which is the same than pattern_id in
-                * the formula *)
-               let pattern_matches_per_id =
-                 group_matches_per_pattern_id res.matches
-               in
+               let dummy_matches = Hashtbl.create 1 in
                let env =
                  {
                    config;
-                   pattern_matches = pattern_matches_per_id;
+                   pattern_matches = dummy_matches;
```

Comment:
Minor comment: We could also create `env` in `evaluate_formula`. It's not that many arguments I think, because `config` is computed from the `rule` and it's only needed in `evaluate_formula`. And `file`, `xlang` and `lazy_ast_and_errors` could be passed together as `file_and_more` (which is already an argument of `check`).

Reply:
And after my PR, I wonder if we should be passing the entire rule in `env` or just the `rule.id` which is the only thing we really use right now!

Reply:
Happy to change it; I wasn't very comfortable with the dummy

Reply:
Actually it is looking like a lot---I'll think about it more but merge for now